### ISSUE TITLE
building `biniou` compiles `easy-format` as a dependency & then fails to build using it

### ISSUE THREAD

**Issue by dysinger** [timbertson/opam2nix-packages#39]
```
➜ nix-shell                            
building '/nix/store/58j4sfy0v2n711n9f6nlbjn0055aq2zc-biniou-1.2.0.drv'...
building '/nix/store/rrck9abbh85m0bd95vnnskfhjar3rzfc-jbuilder-transition.drv'...
unpacking sources
unpacking source archive /nix/store/grmmhc9kma1m2h5pmb2bplki0qsjy0sz-v1.2.0.tar.gz
source root is biniou-1.2.0
setting SOURCE_DATE_EPOCH to timestamp 1500401350 of file biniou-1.2.0/test/test_biniou.ml                                             
patching sources                                                                                                                       
configuring                                                                                                                            
building                                                                                                                               
Loading /nix/store/wg87c36hvgln0yw5gkv49nc7v2xn6a2g-opam                                                                               
creating /nix/store/jp4ivs3i7awhzfkhgwax8yxjana1n026-biniou-1.2.0                                                                      
creating /nix/store/jp4ivs3i7awhzfkhgwax8yxjana1n026-biniou-1.2.0/bin
creating /nix/store/jp4ivs3i7awhzfkhgwax8yxjana1n026-biniou-1.2.0/lib
+ jbuilder build -p biniou -j 1
File "src/jbuild", line 9, characters 14-25:
9 |   (libraries (easy-format))))
                  ^^^^^^^^^^^
Error: Library "easy-format" not found.
Hint: try: dune external-lib-deps --missing -p biniou @install
unpacking sources                                                                                                                      
patching sources                                                                                                                       
configuring                                                                                                                            
findlibPreBuildAction                                                                                                                  
building                                                                                                                               
Loading /nix/store/slfzd1zfg8a2230l403b5ds3a22ikifm-opam                                                                               
creating /nix/store/y2s29cwiwzag5ys0pq2104qwsqicww2q-jbuilder-transition/bin
installing                                                                                                                             
Command failed.                                                                                                                        
Loading /nix/store/slfzd1zfg8a2230l403b5ds3a22ikifm-opam                                                                               
no .install file found!
removing empty dir /nix/store/y2s29cwiwzag5ys0pq2104qwsqicww2q-jbuilder-transition/bin
removing empty dir /nix/store/y2s29cwiwzag5ys0pq2104qwsqicww2q-jbuilder-transition/lib
post-installation fixup                                                                                                                
builder for '/nix/store/58j4sfy0v2n711n9f6nlbjn0055aq2zc-biniou-1.2.0.drv' failed with exit code 1                                     
cannot build derivation '/nix/store/rppffhr2jf2zfji2rnl66k8b2f641pxj-atdgen-runtime-2.0.0.drv': 1 dependencies couldn't be built       
cannot build derivation '/nix/store/k5d5adxfrgl25ipsdrppxkhkyr79s7fd-opam2nix-generated-packages.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/zsv9icc1hz01cbijmhyzfz8iknwcrkjm-opam-selection.nix.drv': 1 dependencies couldn't be built         
error: build of '/nix/store/zsv9icc1hz01cbijmhyzfz8iknwcrkjm-opam-selection.nix.drv' failed                                            
(use '--show-trace' to show detailed location information)
```

... looking at the repo nix/opam files it clearly lists easy-format as a dependency of biniou. not sure what's going on here. I'm giong to hit it again tomorrow.

**Comment by timbertson** [comment]
This is an annoying one, not quite sure what the correct fix is. I used `./single-shell.sh` to get into the env, and the problem is that `$OCAMLPATH` is not set. If you do manually set it (to `/nix/store/3ljghwsxzs7dqh6qc39zh67bqhqm7cy0-easy-format-1.3.1/lib` in my case), then `biniou` compiles.

There's a bunch of weirdness around this:
 - OCAMLPATH is only set from the findlib setup-hook: https://github.com/timbertson/opam2nix-packages/blob/3c69182c5f755c56041982ea858304fa2b7fe33f/repo/overrides/ocamlfind/default.nix#L32-L47
 - there's special casing in opam2nix so that everyone (aside from `ocamlfind` and `conf-*` packages) gets an optional dependency on findlib: https://github.com/timbertson/opam2nix/blob/19c1a5089bd029f4d63f24708f962395b7b40ee2/src/opam_metadata.ml#L403-L407

The problem with biniou is that it (and all of its dependencies) claim not to depend on `findlib`. But they _do_ depend on `$OCAMLPATH` being set up. If something in your selections depended on findlib, everything would probably just work.

I'm not sure what the right solution would be. Maybe it should just be a mandatory dependency on `findlib`, and we sort out any bugs caused by that. On second thoughts, the proper solution is probably introducing some dependency which everyone depends on and which only contains the findlib setup-hooks, because setting `$OCAMLPATH` is not just needed for findlib (it looks like dune reimplements findlib's logic using the same var).

**Comment by timbertson** [comment]
Fixed in 22116de1d6535c3014a5a4bd009c900a77fd36ba. It turned out to not be that bad to split out the non-findlib specific setup hook, and now that's part of `ocaml` so everybody gets it :tada:
